content/docs/digging_deeper/transmit.md

```diff
@@ -0,0 +1,373 @@
+:::
+
+### Configuring the Transmit Instance
+
+The `Transmit` class accepts an object with the following properties:
+
+<dl>
+
+<dt>
+
+baseUrl
+
+</dt>
+
+<dd>
+
+The base URL of the server. The URL must include the protocol (http or https) and the domain name.
+
+</dd>
+
+<dt>
+
+uidGenerator
+
+</dt>
+
+<dd>
+
+A function that generates a unique identifier for the client. The function must return a string. It defaults to `crypto.randomUUID`.
+
+</dd>
+
+<dt>
+
+eventSourceFactory
+
+</dt>
+
+<dd>
+
+A function that creates a new `EventSource` instance. It defaults to the WebAPI `EventSource`. You need to provide a custom implementation if you want to use the client on `Node.js`, `React Native` or any other environment that does not support the `EventSource` API.
+
+</dd>
+
+<dt>
+
+eventTargetFactory 
+
+</dt>
+
+<dd>
+
+A function that creates a new `EventTarget` instance. It defaults to the WebAPI `EventTarget`. You need to provide a custom implementation if you want to use the client on `Node.js`, `React Native` or any other environment that does not support the `EventTarget` API. Return `null` to disable the `EventTarget` API.
+
+</dd>
+
+<dt>
+
+httpClientFactory
+
+</dt>
+
+<dd>
+
+A function that creates a new `HttpClient` instance. It is mainly used for testing purposes.
+
+</dd>
+
+<dt>
+
+beforeSubscribe
+
+</dt>
+
+<dd>
+
+A function that is called before subscribing to a channel. It receives the channel name and the `Request` object sent to the server. Use this function to add custom headers or modify the request object.
+
+</dd>
+
+<dt>
+
+beforeUnsubscribe
+
+</dt>
+
+<dd>
+
+A function that is called before unsubscribing from a channel. It receives the channel name and the `Request` object sent to the server. Use this function to add custom headers or modify the request object.
+
+</dd>
+
+<dt>
+
+maxReconnectAttempts
+
+</dt>
+
+<dd>
+
+The maximum number of reconnection attempts. It defaults to `5`.
+
+</dd>
+
+<dt>
+
+onReconnectAttempt
+
+</dt>
+
+<dd>
+
+A function that is called before each reconnection attempt and receives the number of attempts made so far. Use this function to add custom logic.
+
+</dd>
+
+<dt>
+
+onReconnectFailed
+
+</dt>
+
+<dd>
+
+A function that is called when the reconnection attempts fail. Use this function to add custom logic.
+
+</dd>
+
+<dt>
+
+onSubscribeFailed
+
+</dt>
+
+<dd>
+
+A function that is called when the subscription fails. It receives the `Response` object. Use this function to add custom logic.
+
+</dd>
+
+<dt>
+
+onSubscription
+
+</dt>
+
+<dd>
+
+A function that is called when the subscription is successful. It receives the channel name. Use this function to add custom logic.
+
+</dd>
+
+<dt>
+
+onUnsubscription
+
+</dt>
+
+<dd>
+
+A function that is called when the unsubscription is successful. It receives the channel name. Use this function to add custom logic.
+
+</dd>
+
+</dl>
+
+
+### Creating a Subscription
```

Comment:
An image could help I think cause this could be confusing that the subscription method build a subscription object but does not subscribe.

(CLIENT) Transmit -> transmit -> subscription -> register ====> (SERVER)

Reply:
It is explained below your remark. I have added a Note if it wasn't evident from the explanation.